MSD Backups: add activity type filter to Backups list

Initsogar opened the pull request:
<!--
Link a related Github/Linear issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword. Consider using "part of" instead.
-->

Resolves DOTMSD-762
~Depends on 195500-ghe-Automattic/wpcom~

## Proposed Changes

- Add data-layer for new `/sites/[site]/activity/rewindable/count/group` API endpoint (implemented on 195500-ghe-Automattic/wpcom)
- Add activity type filtering with counts to the Backups list dataviews

## Demo

https://github.com/user-attachments/assets/1d624198-0b3a-4cbf-80c1-90f13a51dc00

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

This PR adds activity type filtering similar to the Activity Log page, but focused on backup-related activities. Users can now quickly filter by activity type (Plugins, Posts and Pages, Themes, etc.) and see how many items exist in each category, making it much easier to navigate large backup lists.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Spin up a Calypso Live Branch
* Navigate to `/v2/sites/[site]/backups`
* Verify the "Type" filter dropdown appears in the DataViews toolbar
* Verify the filter shows activity types with counts (e.g., "Plugins (5)", "Backups (35)")
* Select one or more activity types and verify the list filters correctly
* Test with date range filtering and verify counts update accordingly

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you tested accessibility for your changes? Ensure the feature remains usable with various user agents (e.g., browsers), interfaces (e.g., keyboard navigation), and assistive technologies (e.g., screen readers) (PCYsg-S3g-p2).
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

Initsogar committed:
Add filter by activity type on Backups list

Initsogar committed:
Merge branch 'trunk' into update/msd-backup-filters

Initsogar committed:
Merge branch 'trunk' into update/msd-backup-filters

Initsogar committed:
Preload siteBackupActivityLogGroupCountsQuery

Initsogar committed:
Merge branch 'trunk' into update/msd-backup-filters

keoshi reviewed (approved):
LGTM, @Initsogar!

Not for now or blocking, but the only question I have is whether we should include the counts. Seems useful, but also introduced in a way that is non-standard. However, we don't have any standard way of surfacing these, so maybe it's fine.

@jameskoster Have we ever considered adding counts to the filtering options in DataViews? Rough mock:

Before | After
--|--
<img width="319" height="198" alt="image" src="https://github.com/user-attachments/assets/389abfa1-dd10-41ec-8499-f8d27a1f6703" /> | <img width="638" height="396" alt="image" src="https://github.com/user-attachments/assets/cfd649b2-9285-4ef4-98a5-b396eb763528" />

StevenDufresne commented on client/dashboard/sites/backups/dataviews/fields.tsx:
```diff
@@ -2,13 +2,45 @@ import { Icon } from '@wordpress/components';
+		return '';
+	}
+	const [ group ] = name.split( '__' );
+	return group ?? '';
```
Minor: Does it make sense to just use the `name` as the fallback here?

Initsogar replied:
yep, that make sense. Addressed in https://github.com/Automattic/wp-calypso/pull/106953/commits/306e14f534831106b8988db081370c9f6f1cbd74

StevenDufresne commented on client/dashboard/sites/backups/dataviews/fields.tsx:
```diff
@@ -2,13 +2,45 @@ import { Icon } from '@wordpress/components';
+		? Object.entries( activityLogTypes )
+				.map( ( [ value, { name, count } ] ) => {
+					// Override "Backups and Restores" to just "Backups" for backup list context
+					const displayName = name === 'Backups and Restores' ? __( 'Backups' ) : name;
```
Confirming that `Backups and Restores` won't change regardless of locale or user data, and this string match will always work.

Initsogar replied:
Nice catch. I just updated this to use the value (`rewind`) instead of the label that is translated from the backend.

Addressed in https://github.com/Automattic/wp-calypso/pull/106953/commits/acda790c3aa5f24c05caa242a916a6e2fbed5777

StevenDufresne commented on client/dashboard/sites/backups/dataviews/fields.tsx:
```diff
@@ -2,13 +2,45 @@ import { Icon } from '@wordpress/components';
+					const displayName = name === 'Backups and Restores' ? __( 'Backups' ) : name;
+					return {
+						value,
+						label: `${ displayName } (${ count })`,
```
I believe this will be the only DataView with counts in the filters. UX consistency is really important. 

We should be planning to add it everywhere if we add it here. Thoughts?

fushar replied:
> I believe this will be the only DataView with counts in the filters.

I found it in Logs -> Activity:

<img width="414" height="431" alt="image" src="https://github.com/user-attachments/assets/820b0749-66e0-4e2d-bf76-fe2a80b60960" />

> UX consistency is really important.

I believe it's already discussed above 😄 https://github.com/Automattic/wp-calypso/pull/106953#pullrequestreview-3442801133 (`so maybe it's fine`)

But I totally agree that we should make it consistent everywhere, or at least make it configurable at DataViews level.

StevenDufresne replied:
> I believe it's already discussed above 😄 https://github.com/Automattic/wp-calypso/pull/106953#pullrequestreview-3442801133 (so maybe it's fine)

That's specific to DataViews, not to adding counts to all the dashboard DataViews with the `{label} ({count})` format.

I love this feature. I want it everywhere. Is that selfish of me?

jameskoster replied:
It seems inevitable that this would be useful in other contexts, so it's probably worth surfacing upstream.

Initsogar committed:
Fix activity type slug retrieval to validate by value and not by name

Initsogar committed:
Update activity type slug retrieval to return original name if group is not found

Initsogar committed:
Merge remote-tracking branch 'origin/trunk' into update/msd-backup-filters

Initsogar committed:
Remove count from the activity type filter

Initsogar commented:
Hey! Thanks for the comments. I’m at a meetup this week, so I might take a bit longer to reply.

> Not for now or blocking, but the only question I have is whether we should include the counts. Seems useful, but also introduced in a way that is non-standard. However, we don't have any standard way of surfacing these, so maybe it's fine.

> It seems inevitable that this would be useful in other contexts, so it's probably worth surfacing upstream.

Thanks @keoshi @jameskoster @StevenDufresne @fushar! I initially added the counts in the type filter to keep it similar to the Activity Log page, but since it could create some inconsistency, I’ve removed them from this PR for now. If it is fine to keep them, please let me know. I just wanted to make sure we are not introducing new inconsistencies at this stage of the project.

I don’t have strong opinions about using the counts, but I have to admit that @keoshi’s proposal looks nice:

<img width="638" height="396" alt="image" src="https://github.com/user-attachments/assets/bb612d25-7408-4d5d-b7e6-4a68f775f988" />

@keoshi @jameskoster what would be the next steps for deciding whether to include the counts in DataViews filters?

Initsogar committed:
Merge remote-tracking branch 'origin/trunk' into update/msd-backup-filters
